DOCS-5309 GitLab Integration Edits + Metrics Versioning Organization (#15085)
DataDog/integrations-core: gitlab/README.md

## Setup

### Prerequisites

> This OpenMetrics-based integration has a latest version (V2) and a legacy version (V1). To get all the most up-to-date features, Datadog recommends upgrading to the latest version. For more information, see the [Latest and Legacy Versioning For OpenMetrics-based Integrations][17].

Reviewer: I'm not sure about giving this such a prominent place (its own section and at the beginning of the setup). I also wouldn't think of this as a "prerequisite". What would you think of turning this into some sort of note (perhaps one of those colored `div`s?), probably placed somewhere in the `##### Metric collection` subsection below? Note that there's also a reference to a "legacy example" on that first point as well.

https://github.com/DataDog/integrations-core/blob/86a09fc3836b9145bf97d22ea2fc19c582cff587/gitlab/README.md?plain=1#L32-L35

This is because the most obvious way in which the reader is expected to make use of this information is either when setting it up or when looking for specific metrics that they might be expecting.

Author: We introduce the Prerequisites section across all the integration READMEs 🤔

> Metrics marked as `[Legacy]`, `[OpenMetricsV1]`, or `[OpenMetricsV2]` are only available using the corresponding version of the GitLab integration. Metrics not marked are collected by all three versions.

Reviewer: Is there any way to keep this note closer to where the metrics are listed (ideally, right before them).

Also, I think the term "Legacy" here is becoming overloaded and I think we'll be able to drop it (just opened #15296).

```suggestion
Metrics marked as `[OpenMetricsV1]` or `[OpenMetricsV2]` are only available using the corresponding version of the GitLab integration. All other metrics are collected by both versions.
```

Author: @alopezz We can't move the Metrics sentence in front of the magic sentence in the ## Metrics section otherwise it will not autogenerate the table of metrics unfortunately: https://datadoghq.atlassian.net/wiki/spaces/docs4docs/pages/1169426657/Integrations#metadata.csv.

## Setup

This OpenMetrics-based integration has a latest mode (`use_openmetrics`: true) and a legacy mode (`use_openmetrics`: false). To get all the most up-to-date features, Datadog recommends enabling the latest mode. For more information, see [Latest and Legacy Versioning For OpenMetrics-based Integrations][18].

> Metrics marked as `[OpenMetricsV1]` or `[OpenMetricsV2]` are only available using the corresponding mode of the GitLab integration. All other metrics are collected by both modes.

Author: ```suggestion
Metrics marked as `[OpenMetricsV1]` or `[OpenMetricsV2]` are only available using the corresponding mode of the GitLab integration. All other metrics are collected by both modes. 

To use Prometheus/OpenMetrics v1 instead of OpenMetrics v2, change the `use_openmetrics` option to `use_prometheus`, and change the `openmetrics_endpoint` option to `prometheus_url`. For more information, see the [Prometheus and OpenMetrics metrics collection from a host documentation][30].
```

## Setup

> ### Prerequisites
> 
> This OpenMetrics-based integration has a latest version (V2) and a legacy version (V1). To get all the most up-to-date features, Datadog recommends upgrading to the latest version. For more information, see [Latest and Legacy Versioning For OpenMetrics-based Integrations][18].

Author: ```suggestion
This OpenMetrics-based integration has a latest mode (`use_openmetrics`: true) and a legacy mode (`use_openmetrics`: false). To get all the most up-to-date features, Datadog recommends enabling the latest mode. For more information, see [Latest and Legacy Versioning For OpenMetrics-based Integrations][18].
```

> This OpenMetrics-based integration has a latest version (V2) and a legacy version (V1). To get all the most up-to-date features, Datadog recommends upgrading to the latest version. For more information, see the [Latest and Legacy Versioning For OpenMetrics-based Integrations][18].

Author: ```suggestion
This OpenMetrics-based integration has a latest version (V2) and a legacy version (V1). To get all the most up-to-date features, Datadog recommends upgrading to the latest version. For more information, see [Latest and Legacy Versioning For OpenMetrics-based Integrations][18].
```

> This OpenMetrics-based integration has a latest mode (`use_openmetrics`: true) and a legacy mode (`use_openmetrics`: false). To get all the most up-to-date features, Datadog recommends enabling the latest mode. For more information, see [Latest and Legacy Versioning For OpenMetrics-based Integrations][18].

Reviewer: Another `openmetrics_endpoint` switch like https://github.com/DataDog/integrations-core/pull/15073#discussion_r1317050535

Author: ```suggestion
This OpenMetrics-based integration has a latest mode (enabled by setting `openmetrics_endpoint` to point to the target endpoint) and a legacy mode (enabled by setting `prometheus_url` instead)). To get all the most up-to-date features, Datadog recommends enabling the latest mode. For more information, see [Latest and Legacy Versioning For OpenMetrics-based Integrations][18].
```

> Metrics marked as `[OpenMetricsV1]` or `[OpenMetricsV2]` are only available using the corresponding version of the GitLab integration. All other metrics are collected by both versions.

Author: ```suggestion
Metrics marked as `[OpenMetricsV1]` or `[OpenMetricsV2]` are only available using the corresponding mode of the GitLab integration. All other metrics are collected by both versions.
```

> To use Prometheus/OpenMetrics v1 instead of OpenMetrics v2, change the `use_openmetrics` option to `use_prometheus`, and change the `openmetrics_endpoint` option to `prometheus_url`. For more information, see the [Prometheus and OpenMetrics metrics collection from a host documentation][30].

Reviewer: This is related to my other comment. It's fine if we leave this information as a separate paragraph (though I think it should go right after the explanation about "modes"), but we should replace the references from v1 and v2 to legacy and latest like I've suggested in a different PR with a similar paragraph.

Author: ```suggestion
To use the legacy mode of Prometheus/OpenMetrics instead of the latest mode of OpenMetrics, change the `use_openmetrics` option to `use_prometheus`, and change the `openmetrics_endpoint` option to `prometheus_url`. For more information, see the [Prometheus and OpenMetrics metrics collection from a host documentation][30].
```

[13]: https://github.com/DataDog/integrations-core/blob/master/gitlab/assets/service_checks.json
[14]: https://docs.datadoghq.com/help/
[15]: https://docs.gitlab.com/ee/user/admin_area/monitoring/health_check.html#readiness
[16]: https://github.com/DataDog/integrations-core/blob/7.43.x/gitlab/datadog_checks/gitlab/data/conf.yaml.example

> [17]: https://docs.datadohgq.com/integrations/guide/versions-for-openmetrics-based-integrations

Reviewer 2: ```suggestion
[17]: https://docs.datadoghq.com/integrations/guide/versions-for-openmetrics-based-integrations
```

But does this page exist?
